event.go

```diff
@@ -110,40 +132,51 @@ func NewEvent(payload interface{}, err error) Event {
 	return event
 }
 
-func (e *Event) NewEvent(payload interface{}, err error) Event {
-	event := Event{
-		ID:           uuid.NewV4(),
-		ParentID:     e.ID,
-		Name:         name(payload),
-		Payload:      payload,
-		PayloadModel: reflect.TypeOf(payload).String(),
-		Error:        err,
-		CreatedAt:    time.Now(),
-	}
+func (e *Event) CreateEvent(action Action, state State, stateMessage string) Event {
```

Comment:
We need the action here to change it to status, failed or complete when responding back

Reply:
we dont use the action when createing a NewEvent because we pre set it to waiting. I think it's pretty clear when using `transistor.CreateEvent` and `e.CreateEvent` that you are creating a new and responding to.